### PR TITLE
Change dropdowns to always keep options in the DOM

### PR DESCRIPTION
**Pull request by rekabmot**
# Description

Currently, dropdown selects only render the options list if the
component is focused. This makes testing applications using browser
automation difficult as these options cannot be selected.

This PR ensures that the options are always rendered within the DOM, but
makes them hidden unless the component is focused.

# TODO
- [x] Release notes

* Note that a lot of the changes are fixes for associated tests where an array containing a single blank JS object was being passed as the options array.

### CODE REVIEW

**Comment by andrewjtait** [comment]
We already have automation which tests over these components without any issue - do we need this change?

File: src/components/dropdown/dropdown.scss [hunk 1]
```diff
@@ -16,6 +16,10 @@ $dropdown-input-icon-width: 20px !default;
   }
 }
 
+.carbon-dropdown__list-hidden {
+  display: none;
```
Comment (toni-leigh) [inline src/components/dropdown/dropdown.scss]: Are we sure about `display: none` - it also removes the things from the DOM

Will automation ever process css? Also, are their accessibility considerations here?

**Comment by rekabmot** [comment]
@andrewjtait @toni-leigh The application using these components uses Sahi for automation, which de-focuses elements in between each action. When a dropdown is deselected, all it's options are removed from the DOM, which makes it impossible for Sahi to find them. With this change, the elements are still in the DOM and automation is able to select them even with the dropdown closed.

In terms of accessibility, I'm not an expert but I would imagine having the options existing at all times would make it easier for a screen reader to find them? I was wondering about the bigger picture of accessibility the other day anyway, as Carbon doesn't appear to match traditional HTML elements semantically - but that's another conversation :)

The same reasoning was used on this previously merged PR which addressed the same issue with another component: https://github.com/Sage/carbon/pull/1203

**Comment by toni-leigh** [comment]
@rekabmot I approve of this change because generally I think we should match the HTML elements as closely as possible - there's a lot of iterations and years and years of development that have gone into these things and they're considering use cases we just don't even think of, including accessibility. I read an article once (which I can't find) detailing _a lot_ different ways of a human interacting with a drop-down - and that's before you consider the robots!

**Comment by SeanArmstrong** [comment]
@toni-leigh I think this may be a better solution -

https://github.com/Sage/carbon/blob/4d09fe9e844ac97b529230a6276c6ffc0e6add17/src/style-config/mixins.scss#L244-L251

However this is not merged yet :(. You can probably just copy over the code for now and put a comment in

**Comment by toni-leigh** [comment]
@SeanArmstrong what's the PR that mixin is associated with? Could just cut it from there and paste it here, given this is going to merged soon as this is blocking an app - that PR would have to merge after this has gone though?

**Comment by SeanArmstrong** [comment]
@toni-leigh Sorry i meant to include that too - https://github.com/Sage/carbon/pull/1248/files

**Comment by toni-leigh** [comment]
I suspect that as the table header thing is a backlog ticket and this is blocking someone, this will go first!

**Pull request merged**